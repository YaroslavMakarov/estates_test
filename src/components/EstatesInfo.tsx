import React from 'react';
import { useParams } from 'react-router-dom';
import { Row, Col } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useSelector } from 'react-redux';

import BackButton from '../components/BackButton';
import EstateInfoCarousel from '../components/EstateInfoCarousel';
import NoInfo from '../components/NoInfo';

import { Img, EstateDescription } from '../styled/estateStyles';
import { estatesSelector } from '../redux/estates';

type Params = {
    id: string;
};

const EstateInfo = () => {
const params: Params = useParams();
const estates = useSelector(estatesSelector);

const currentEstate = estates.filter(estate => estate.id === Number(params.id));

if (currentEstate.length === 0) {
    return (
        <NoInfo />
    );
}

    return (
        <>
            <BackButton />
            <Row>
                <Col sm={5}>
                    <Img 
                        src={currentEstate[0].preview_img}
                        id={params.id}
                    />
                </Col>
                <Col sm={{offset: 1, span: 5}}>
                    <EstateDescription id={params.id}>
                        {`Name: ${currentEstate[0].title}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Price: ${currentEstate[0].price}$`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Address: ${currentEstate[0].address.street}, ${currentEstate[0].address.number}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Seller: ${currentEstate[0].seller}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Description: ${currentEstate[0].description}`}
                    </EstateDescription>
                </Col>
            </Row>
            <EstateInfoCarousel photos={currentEstate[0].detailed_photos} />
        </>
    )
};

export default EstateInfo;