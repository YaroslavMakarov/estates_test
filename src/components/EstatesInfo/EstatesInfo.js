import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { Container, Row, Col, Button } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useSelector } from 'react-redux';

import EstateInfoCarousel from '../EstateInfoCarousel/EstateInfoCarousel';

import { Img, EstateDescription } from '../../styled/estateStyles';
import { estatesSelector } from '../../redux/estates';

const EstateInfo = () => {
const params = useParams();
const history = useHistory();
const estates = useSelector(estatesSelector);

const currentEstate = estates.find(estate => {
    if (estate.id === Number(params.id)) {
        return estate;
    }
})

    return (
        <>
            <Container
              fluide
              className="p-0"
            >
                <Button
                variant="dark"
                className="mb-3"
                onClick={() => history.goBack()}
                >
                    Back
                </Button>
            </Container>
            <Row>
                <Col sm={5}>
                    <Img 
                        src={currentEstate.preview_img}
                        id={params.id}
                    />
                </Col>
                <Col sm={{offset: 1, span: 5}}>
                    <EstateDescription id={params.id}>
                        {`Name: ${currentEstate.title}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Price: ${currentEstate.price}$`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Address: ${currentEstate.address.street}, ${currentEstate.address.number}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Seller: ${currentEstate.seller}`}
                    </EstateDescription>
                    <EstateDescription id={params.id}>
                        {`Description: ${currentEstate.description}`}
                    </EstateDescription>
                </Col>
            </Row>
            <EstateInfoCarousel photos={currentEstate.detailed_photos} />
        </>
    )
};

export default EstateInfo;