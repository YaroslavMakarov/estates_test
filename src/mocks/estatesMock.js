export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};