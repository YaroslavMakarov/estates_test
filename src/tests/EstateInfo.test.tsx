import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import EstatesInfo from '../components/EstatesInfo';
import { estatesMock } from '../mocks/estatesMock';
const mockStore = configureStore();

describe('EstateInfo component', () => {
    let wrapper: RenderResult;
    let estate: Estate;
    const mockedStore = mockStore(estatesMock);

    describe('EstateInfo without info', () => {
        beforeEach(() => {
            wrapper = render(  
                <Provider store={mockedStore}>
                    <MemoryRouter initialEntries={["/info/3"]}>
                        <Route path="/info/:id" component={EstatesInfo} />
                    </MemoryRouter>
                </Provider>    
            );
        });

        it('get snapShot estates info without estates', () => {
            expect(wrapper.baseElement).toMatchSnapshot();
        });
    });

    describe('EstateInfo with info', () => {
        beforeEach(() => {
            estate = estatesMock.estates.estates[0];

            wrapper = render(  
                <Provider store={mockedStore}>
                    <MemoryRouter initialEntries={["/info/1"]}>
                        <Route path="/info/:id" component={EstatesInfo} />
                    </MemoryRouter>
                </Provider>    
            );
        });

        it('get snapShot estates info without estates', () => {
            expect(wrapper.baseElement).toMatchSnapshot();
        });

        it('check Img attr', () => {
            expect(screen.getByAltText('main img')).toHaveAttribute('src', `${estate.preview_img}`);
        });

        it('check EstateInfo name', () => {
            expect(screen.getByText(`Name: ${estate.title}`)).toBeTruthy();
        });
    
        it('check EstateInfo pricce', () => {
            expect(screen.getByText(`Price: ${estate.price}$`)).toBeTruthy();
        });
    
        it('check EstateInfo address', () => {
            expect(screen.getByText(`Address: ${estate.address.street}, ${estate.address.number}`)).toBeTruthy();
        });

        it('check EstateInfo saller', () => {
            expect(screen.getByText(`Seller: ${estate.seller}`)).toBeTruthy();
        });
    
        it('check EstateInfo description', () => {
            expect(screen.getByText(`Description: ${estate.description}`)).toBeTruthy();
        });
    });
    
});