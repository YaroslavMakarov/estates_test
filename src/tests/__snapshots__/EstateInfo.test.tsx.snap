// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstateInfo component EstateInfo with info get snapShot estates info without estates 1`] = `
<body>
  <div>
    <div
      class="p-0 container-fluid"
    >
      <button
        class="mb-3 btn btn-dark"
        type="button"
      >
        Back
      </button>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-5"
      >
        <img
          alt="main img"
          class="sc-gsTCUz hUELus"
          id="1"
          src="https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg"
        />
      </div>
      <div
        class="col-sm-5 offset-sm-1"
      >
        <div
          class="sc-dlfnbm kGxffL"
          id="1"
        >
          Name: Shevchenko estate
        </div>
        <div
          class="sc-dlfnbm kGxffL"
          id="1"
        >
          Price: 30000$
        </div>
        <div
          class="sc-dlfnbm kGxffL"
          id="1"
        >
          Address: Melnychna,
                                   12
        </div>
        <div
          class="sc-dlfnbm kGxffL"
          id="1"
        >
          Seller: Chack Norris
        </div>
        <div
          class="sc-dlfnbm kGxffL"
          id="1"
        >
          Description: The best from Chuck
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-6 offset-sm-3"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li />
            <li />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg"
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg"
              />
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="slide"
                class="sc-gsTCUz esHetz"
                src="https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
              />
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`EstateInfo component EstateInfo without info get snapShot estates info without estates 1`] = `
<body>
  <div>
    <div
      class="d-flex align-items-center justify-content-center container"
    >
      <div
        class="sc-hKgILt gmArwI"
      >
        Sorry, don't have any information about this estate.
      </div>
    </div>
  </div>
</body>
`;
